engine/class_modules/sc_demon_hunter.cpp

```diff
@@ -4575,6 +4578,12 @@ struct soul_cleave_t : public demon_hunter_attack_t
         m *= 1.0 + p()->talent.vengeance.focused_cleave->effectN( 1 ).percent();
       }
 
+      if ( p()->buff.soul_furnace->up() )
+      {
+        m *= 1.0 + p()->buff.soul_furnace->check_value();
+        p()->buff.soul_furnace->decrement();
```

Comment:
Decrement probably needs to be put in the ability execute() after the base execute() call. This function could be called multiple times potentially.